### DISCUSSION TITLE
Split the logic into 2 different options: "Speedtech Required" and "Standard"

### DISCUSSION THREAD

**Discussion by Myth197** [bordplate/rac1-randomizer#13] [Improvement- and feature-suggestions]
Currently the logic is analogous to "Glitched Logic" found in most other randomizers. I propose the addition of a "Standard" logic with an initial attempt at the logic here:
[Casual Logic.txt](https://github.com/bordplate/rac1-randomizer/files/8263966/Casual.Logic.txt).

Just a little bit of information on who I am as context, I have been playing various zelda randomizers since 2017 at varying levels of involvement at the development level. I am someone who started out with a casual interest in enjoying the games with the slight twist that rando gave, to eventually becomming a full fledged main game speedrunner, community moderator and content creator (The Legend of Zelda: The Minish Cap). I am an advocate for randomizers being an avenue for new players to "have a go" at speedrunning if they innitially had reservations in advance that it "wouldn't be for them" or "they would never be good enough", which is what also held me back from giving it a go for many years. (BTW I've been wanting to learn RAC speedrunning for years and finally took the plunge now there is a randomizer to jump in at, I have no experience speedrunning any ratchet game prior to march 2022 despite being an observer for at least 5 years).

Anyway, with that out of the way, onto some discussion points I would like to make in favour of establishing these two logic options:

Firstly, only people inside the speedrun community are able to complete seeds that have logical requirements to do movement tech that is beyond the knowledge of a casual player. This imediately sets a barrier to entry to the randomizer which prevents it from spreading and growing in popularity. Once players start with the randomizer and are able to do a few seeds, some may have had enough, but others will want to learn how to go faster and will learn the speedtech and become regular runners of the games. Randomizers are an under-recognised entry point for many speedgames.

Secondly, "Why is the easy mode the standard? We'll just move onto the harder logic eventually right?" ; with a strict logic that reflects everything that is possible for the player to acheive, the player has to collect the items in the order that the randomizer places them, one item leads to the next and the seed is beaten in a linear progression. At most there are forced decisions of "Do I go to option A, or to option B?", but there is always the guarantee that the item required to progress will be at one or the other.

However, if we purposefully simplify the logical requirements, then the player is able to do things the randomizer doesn't expect they should be able to do. This leads to "Out of Logic" play, the player can essentially jump ahead in the seed to collect an item they were only "supposed" to get later in the seed. This is a very popular staple of many randomizers and adds another dimension to how you play through the randomizer. The player has to ask themselves if they want to venture "Out of Logic" to get an item, then they can use this as information gathering for figuring out the overall structure of the seed from more viewpoints.

So instead of thinking of these two different options as "Hard" or "Easy", I think its better to think of them as:

- "Speedtech Required": A challenge seed, the randomizer tests if you are able to perform the movement-tech required to reach certain locations in order to beat the seed.
- "Standard": A more open seed, new players are guaranteed to be able to beat the seeds while being hinted at "Out of Logic" locations on their tracker being reachable with techniques they can learn for the future. Meanwhile expert players try to "Break" the seed by going out of logic, attempting to possibly skip parts of the seed and beat it as fast as possible.

Now for a little speculation on what I think racing would look like on either logic type. "Speedtech" races I think would be similar to races of the regular game, it comes down to how good the player is at performing speedtech. Every now and then you get decisions of going to one place or another which can throw the runners some curveballs, but I believe in general for these seeds its still heavily down the the runners executions. For "Standard" races, there is going to be more testing of how good of a "Rando player" the runner is, many things factor into this such as how quickly can they evaluate the relative time costs of one location over another on the fly? Of course execution will still remain a major factor of deciding who the winner will be, but it won't always be a forgone conclusion from the get go.

That last part is very speculative since I haven't played the randomizer as of writing this (I will update this discussion with some thoughts eventually after testing both the logic options), so these are my initial thoughts. Of course I shouldn't be the final decider of the direction this randomizer goes, as a relative outsider I simply wish to advise and provide information.
This is up to the community to decide what kind of randomizer they want, and what kind of randomizer they want to see raced.

**Comment by bordplate** [comment]
Thank for you for writing this, it's explained really well and I appreciate that. I agree with what you're saying and will definitely split the logic the way you're suggesting. As I said on Discord, I'll do it once I've added the weapons to the items randomized pool.

Thanks for the `Casual Logic.txt` as well, that's super helpful. And just thanks for all the work you've put into helping me with this, I'm extremely thankful.

**Comment by Myth197** [comment]
Just adding to this, eventually it would be good to add a fully customizable logic system. Players can toggle what locations are randomized, and for those also toggle which strats they would like to possibly be required for that location. In which case its probably for the best to add an extra layer to the logic system:

Locations have Strat requirements, then Strats have Item requirements. Each location will have its "Intended" Strat that has the item requirements from the casual logic, but then every strat in the game has its item requirements. Eg: ILJ requires Helipack, ISF requires Thruster, Itemless requires nothing

Then the logic can be written entirely with or statements for the strats, Eg: Aridia trespasser location has 4 strats associated to it:
Intended (Swingshot)
ISF (Thruster)
ILJ (Heli)
PDA IJ (PDA)

This can help make the logic a little more readable, but will help in the future making adding future locations much easier and being able to toggle the local logic per strat easier to implement.

**Comment by Myth197** [comment]
Been in for a while now, probably back in this: #25
